internal/api/v20240610preview/hcpopenshiftclusters_methods.go

```diff
@@ -319,45 +318,53 @@ func (c *HcpOpenShiftClusterResource) Normalize(out *api.HCPOpenShiftCluster) {
 // contains structured but user-friendly details for all discovered errors.
 func validateStaticComplex(normalized *api.HCPOpenShiftCluster) []arm.CloudErrorBody {
 	var errorDetails []arm.CloudErrorBody
-	//Generate a slice of Resource IDs of ControlplaneOperators MI only , disregard the DataPlaneOperators.
-	var cpmiResourceIDs []string
+
 	if normalized.Identity.UserAssignedIdentities != nil {
+		//Initiate the map that will have the number occurence of resource IDs in the Identity and Operator Authn fields .
+		identityOccurrences := make(map[string]int)
+		operatorAuthnOccurrences := make(map[string]int)
+		//Generate a Map of Resource IDs of ControlplaneOperators MI and Service MI if exists , disregard the DataPlaneOperators.
 		for _, operatorResourceID := range normalized.Properties.Spec.Platform.OperatorsAuthentication.UserAssignedIdentities.ControlPlaneOperators {
-			cpmiResourceIDs = append(cpmiResourceIDs, operatorResourceID)
+			operatorAuthnOccurrences[operatorResourceID]++
 		}
 		smiResourceID := normalized.Properties.Spec.Platform.OperatorsAuthentication.UserAssignedIdentities.ServiceManagedIdentity
-		miCount := len(normalized.Identity.UserAssignedIdentities)
-		for resourceID, _ := range normalized.Identity.UserAssignedIdentities {
-
-			if slices.Contains(cpmiResourceIDs, resourceID) || resourceID == smiResourceID {
-				//Duplicated Identity Validation
-				if slices.Contains(cpmiResourceIDs, resourceID) && resourceID == smiResourceID {
-					errorDetails = append(errorDetails, arm.CloudErrorBody{
-						Message: fmt.Sprintf(
-							"%s can not be used in both controlPlaneOperators and serviceManagedIdentity",
-							resourceID),
-						Target: "properties.spec.platform.operatorsAuthentication.userAssignedIdentities",
-					})
-				}
-			} else {
+
+		if smiResourceID != "" {
+			operatorAuthnOccurrences[smiResourceID]++
+		}
+
+		for resourceID := range normalized.Identity.UserAssignedIdentities {
+			identityOccurrences[resourceID]++
+		}
+		for resourceID := range operatorAuthnOccurrences {
```

Comment:
Suggest iterating over `normalized.Properties.Spec.Platform.OperatorsAuthentication.UserAssignedIdentities.ControlPlaneOperators` here.  That way you'll have the operator name to use in the `Target` field:
```
Target: fmt.Sprintf("properties.spec.(blah, blah).controlPlaneOperators[%s]", mapKey)
```

It does mean you'll have to repeat some checks for `ServiceManagedIdentity` (or maybe split the checks into a separate function).